Add decription for new ArangoSearch view properties (3.4) (#7437)
arangodb/arangodb: Documentation/Books/Manual/Views/ArangoSearch/DetailedOverview.md

  > reflected by queries invoked in subsequent ArangoDB transactions, while
  > in-progress ArangoDB transactions will still continue to return a
  > repeatable-read state.
  
  
  ArangoSearch performs operations in its index based on numerous writer
  objects that are mapped to processed segments. In order to control memory that
  is used by these writers (in terms of "writers pool") one can use

>   `writebuffer*` properties of a view.

Reviewer: This block is rendered as expected, but is indented in source. I guess this works because of the two blank lines above.
Should be okay, but I would remove the leading two spaces from the lines to better reflect the intention.

- **writebufferIdle** (_optional_; type: `integer`; default: `64`;
  to disable use: `0`)

  Maximum number of writers (segments) cached in the pool.

- **writebufferActive** (_optional_; type: `integer`; default: `0`;
  to disable use: `0`)

>   Maximum number of concurrent active writers (segments) performs (a transaction).

Reviewer: Why is `(a transaction)` in parentheses? It's not a full sentence without it.

Author: Fixed.

>   Other writers (segments) are wait till current active writers (segments) finish.

Reviewer: ```suggestion
  Other writers (segments) wait till current active writers (segments) finish.
```

Author: Fixed.

- **writebufferSizeMax** (_optional_; type: `integer`; default: `33554432`;
  to disable use: `0`)

  Maximum memory byte size per writer (segment) before a writer (segment) flush is

>   triggered. `0` value turns off this limit fon any writer (buffer) and will be

Reviewer: ```suggestion
  triggered. `0` value turns off this limit for any writer (buffer) and will be
```

>   flushed only after a period [defined](../../Programs/Arangod/Server.md#data-source-flush-synchronization)

Reviewer: ```suggestion
  flushed only after a
```

Author: Fixed.

>   for the flush thread during ArangoDB server startup. `0` value should be used 

Reviewer: ```suggestion
  [period defined for the flush thread](../../Programs/Arangod/Server.md#data-source-flush-synchronization)
  during ArangoDB server startup. `0` value should be used 
```
Is the meaning that the flush period only be set on server startup, or that flushing only occurs during server startup? We might need to tweak the sentence a bit, to avoid misinterpretation.

Author: Fixed.

  > in-progress ArangoDB transactions will still continue to return a
  > repeatable-read state.

> 

Reviewer: "ArangoSearch performs..." (below) is meant as description for the following properties, and not related to the `consolidationIntervalMsec` option, right? I think it would be better in this case to have this as normal text and start another list below, which is actually a continuation of above list.

>   > ArangoSearch performs operations in its index based on numerous writer

Reviewer: ```suggestion
ArangoSearch performs operations in its index based on numerous writer
```

>   > objects that mapped to processed segments. In order to control memory that

Reviewer: Not sure, but I believe an `are` is missing here?
```suggestion
  > objects that are mapped to processed segments. In order to control memory that
```

>   > objects that mapped to processed segments. In order to control memory that

Reviewer: ```suggestion
objects that mapped to processed segments. In order to control memory that
```

>   > objects that mapped to processed segments. In order to control memory that

Reviewer: ```suggestion
objects that mapped to processed segments. In order to control memory that
```

>   > is used by these writers (in terms of "writers pool") one can use

Reviewer: ```suggestion
is used by these writers (in terms of "writers pool") one can use
```

>   > `writebuffer*` properties of a view.

Reviewer: ```suggestion
`writebuffer*` properties of a view.
```

- **writebufferIdle** (_optional_; type: `integer`; default: `64`;
  to disable use: `0`)

>   Maximum number of writers (segments) cached in the pool. `0` value turns off

Reviewer: It already says `to disable use: 0` above, can we therefore remove the sentence that repeats this here?

  caching.

- **writebufferActive** (_optional_; type: `integer`; default: `0`;
  to disable use: `0`)

  Maximum number of concurrent active writers (segments) performs (a transaction).
  Other writers (segments) are wait till current active writers (segments) finish.

>   `0` value turns off this limit and used by default. 

Reviewer: De-duplicate info? If no, please correct:

```suggestion
  `0` value turns off this limit and is used by default. 
```

- **writebufferSizeMax** (_optional_; type: `integer`; default: `33554432`;
  to disable use: `0`)

  Maximum memory byte size per writer (segment) before a writer (segment) flush is
  triggered. `0` value turns off this limit fon any writer (buffer) and will be

>   flushed only after a period defined for special thread during ArangoDB server

Reviewer: What is "special thread"? Is this an implementation detail the user needs to know at all?
Might be missing an "is":
```suggestion
  flushed only after a period defined for a special thread during ArangoDB server
```

    upon several possible configurable formulas as defined by their types.
    The currently supported types are:

>     - **bytes_accum**: Consolidate is performed based on current memory cunsumption

Reviewer: ```suggestion
    - **bytes_accum**: Consolidation is performed based on current memory consumption
```

Author: Fixed.

      of segments and `threshold` property value.
    - **tier**: Consolidate based on segment byte size and live document count
      as dictated by the customization attributes.

### `consolidationPolicy` properties for `bytes_accum` type
  - **threshold** (_optional_; type: `float`; default: `0.1`)

    Defines threshold value of `[0.0, 1.0]` possible range. Consolidation is

>     performed when then the sum of all candidate segment's byte

Reviewer: Is "candidate segments" actually plural here? Then it should be `candidate segments'`

Author: Fixed.

>     size is less than the total segment byte size multiplied by the `threshold`.

Reviewer: ```suggestion
    size is less than the total segment byte size multiplied by the `threshold`;
```

Author: Fixed.

>     I.e. the following formula is applied:

Reviewer: ```suggestion
    i.e. the following formula is applied:
```

Author: Fixed.

    `{threshold} > (segment_bytes + sum_of_merge_candidate_segment_bytes) / all_segment_bytes`.

### `consolidationPolicy` properties for `tier` type

  - **segmentsMin** (_optional_; type: `integer`; default: `1`)

    The minimum number of segments that will be evaluated as candidates for consolidation.

  - **segmentsMax** (_optional_; type: `integer`; default: `1`)

    The maximum number of segments that will be evaluated as candidates for consolidation.

  - **segmentsBytesMax** (_optional_; type: `integer`; default: `5368709120`)

    Maxinum allowed size of all consolidated segments in bytes.

  - **segmentsBytesFloor** (_optional_; type: `integer`; default: `2097152`)

    Defines the value (in bytes) to treat all smaller segments as equal for consolidation
    selection.

  - **lookahead** (_optional_; type: `integer`; default: `18446744073709552000`)

    The number of additionally searched tiers except initially chosen candidates based on
    `segmentsMin`, `segmentsMax`, `segmentsBytesMax`, `segmentsBytesFloor` with

>     respect to defined values. Default value falls to search through all existing segments.

Reviewer: ```suggestion
    respect to defined values. Default value falls through to search all existing segments.
```

Author: Fixed.

    - **tier**: Consolidate based on segment byte size and live document count
      as dictated by the customization attributes.

### `consolidationPolicy` properties for `tier` type

>   - **min\_segments** (_optional_; type: `integer`; default: `1`)

Reviewer: The backslash escape for the underscore shouldn't be necessary here. It is not used above for the `bytes_accum` property, and that one is in for quite some time and doesn't seem to cause any trouble:
https://docs.arangodb.com/3.4/Manual/Views/ArangoSearch/DetailedOverview.html#view-properties

```suggestion
  - **min_segments** (_optional_; type: `integer`; default: `1`)
```

    The minimum number of segments that will be evaluated as candidates for consolidation.

>   - **max\_segments** (_optional_; type: `integer`; default: `1`)

Reviewer: ```suggestion
  - **max_segments** (_optional_; type: `integer`; default: `1`)
```

    The maximum number of segments that will be evaluated as candidates for consolidation.

  - **segments_bytes_max** (_optional_; type: `int`; default: `5368709120`)

    Maxinum allowed size of all consolidated segments in bytes.

  - **segments_bytes_floor** (_optional_; type: `int`; default: `2097152`)

    Defines the value (in bytes) to treat all smaller segments as equal for consolidation
    selection.

  - **lookahead** (_optional_; type: `integer`; default: `18446744073709552000`)

>     The number of additionally searched tiers except initially chosen candidated based on

Reviewer: Typo?
```suggestion
    The number of additionally searched tiers except initially chosen candidates based on
```

    `min_segments`, `max_segments`, `segments_bytes_max`, `segments_bytes_floor` with
    respect to defined values. Default value falls to `integer_traits<size_t>::const_max`

>     (in C++ source code).

Reviewer: Is it relevant for the user? What does it actually mean?
